internal/tiltfile/tiltfile_state.go

```diff
@@ -616,12 +622,59 @@ func (s *tiltfileState) assembleK8sV2() error {
 				r.name = opts.newName
 				s.k8sByName[r.name] = r
 			}
+
+			selectors := make([]k8sObjectSelector, len(opts.objects))
+			for i, o := range opts.objects {
+				s, err := selectorFromString(o)
+				if err != nil {
+					return errors.Wrapf(err, "Error making selector from string '%s'", o)
+				}
+				selectors[i] = s
+			}
+
+			claims := make(map[k8s.K8sEntity]claimPair)
+			for i, sel := range selectors {
+				for _, r := range s.k8s {
+					resourcedMatches := filterEntitiesBySelector(r.entities, sel)
+					if len(resourcedMatches) > 0 {
+						return fmt.Errorf("object %s already belongs to resource %s", opts.objects[i], r.name)
+					}
+				}
+
+				unresourcedMatches := filterEntitiesBySelector(s.k8sUnresourced, sel)
+				if len(unresourcedMatches) != 1 {
+					errMsg := fmt.Sprintf("Found %d matches for %s in remaining YAML. Object must match exactly 1 entity", len(unresourcedMatches), opts.objects[i])
+					if len(s.k8sUnresourced) == 0 {
+						errMsg = fmt.Sprintf("%s. All YAML already belongs to a resource", errMsg)
+					} else if len(unresourcedMatches) == 0 {
+						namesOfAvailableUnresourcedYAML := make([]string, len(s.k8sUnresourced))
+						for i, e := range s.k8sUnresourced {
+							namesOfAvailableUnresourcedYAML[i] = selectorStringFromK8sEntity(e)
+						}
+						errMsg = fmt.Sprintf("%s. Available YAML: %s", errMsg, strings.Join(namesOfAvailableUnresourcedYAML, ", "))
+					}
+
+					return errors.New(errMsg)
+				}
+
+				entity := unresourcedMatches[0]
+				if existingClaim, ok := claims[entity]; ok {
+					return fmt.Errorf("%s tried to claim %s but it was already claimed by %s via %s", opts.objects[i], entity.Name(), existingClaim.resource.name, existingClaim.selectorString)
```

Comment:
This object (fullname) was passed to both k8s_resource foo (as X) and k8s_resource foo2 (as Y).

Reply:
leave a TODO about the eventual desired behavior?

Reply:
(but agree that the error message as is is v confusing, i like your proposal)